src/Controller/RepoController.php

```diff
@@ -143,21 +144,22 @@ public function downloads(Request $request, Organization $organization): JsonRes
      *      requirements={"domain"="%domain%","package"="%package_name_pattern%","sep1"="%organization_separator%","sep2"="%domain_separator%"})
      * @Cache(public=false)
      */
-    public function providerV2(Request $request, Organization $organization, string $package): JsonResponse
+    public function providerV2(Request $request, Organization $organization, string $package): Response
     {
-        [$lastModified, $providerData] = $this->packageManager->findProviders(
+        [$lastModified, $loader] = $this->packageManager->findProviders(
             $organization->alias(),
             [new PackageName('', $package)]
         );
 
-        if ($providerData === []) {
+        if ($lastModified === null) {
             throw new NotFoundHttpException();
         }
 
-        $response = (new JsonResponse($providerData))
+        $response = (new StreamedResponse(function () use ($loader): void {
+            echo \json_encode($loader(), JsonResponse::DEFAULT_ENCODING_OPTIONS);
```

Comment:
Alternatively, this could use a stream output like `$outputStream = \fopen('php://output', 'wb');`.